test/rotation_tests.jl

```diff
@@ -0,0 +1,238 @@
+using Base.Test
+using FixedSizeArrays
+using Quaternions
+using Rotations
+
+verbose_rot_test = false
+
+import Rotations: numel
+numel{T <: Rotations.RotationTypes}(X::T) = Rotations.numel(Rotations.strip_eltype(T))
+numel{T}(X::T) = (T <: FixedSizeArrays.FixedArray) || (T <: FixedSizeArrays.AbstractArray)  ? length(X) : error("numel undefined for input of type $(T)")
+
+# a macro to test if the contents of two containers are approximatley equal
+macro contents_approx_eq(a, b)
+    quote
+        n = numel($(esc(a)))
+        @test n == numel($(esc(b)))
+        for i = 1:n
+            ai, bi = getindex($(esc(a)), i), getindex($(esc(b)), i)
+            @test typeof(ai) == typeof(bi)
+            @test_approx_eq ai bi
+        end
+    end
+end
+macro contents_approx_eq_eps(a, b, eps)
+    quote
+        n = numel($(esc(a)))
+        @test n == numel($(esc(b)))
+        for i = 1:n
+            ai, bi = getindex($(esc(a)), i), getindex($(esc(b)), i)
+            @test typeof(ai) == typeof(bi)
+            @test_approx_eq_eps ai bi $(esc(eps))
+        end
+    end
+end
+
+
+# a macro to test if tow types are approximatey equal
+macro types_approx_eq(a, b)
+    quote
+        @test typeof($(esc(a))) == typeof($(esc(b)))
+        @contents_approx_eq($(esc(a)), $(esc(b)))
+    end
+end
+macro types_approx_eq_eps(a, b, eps)
+    quote
+        @test typeof($(esc(a))) == typeof($(esc(b)))
+        @contents_approx_eq_eps($(esc(a)), $(esc(b)), $(esc(eps)))
+    end
+end
+
+# a macro to test if the contents of two containers are approximatley equal, without examining the element types
+macro contents_approx_eq_notype(a, b)
+    quote
+        n = numel($(esc(a)))
+        @test n == numel($(esc(b)))
+        for i = 1:n
+            ai, bi = getindex($(esc(a)), i), getindex($(esc(b)), i)
+            @test_approx_eq ai bi
+        end
+    end
+end
+
+
+# build a full list of rotation types including the different ordering schemas
+rot_types = Vector{Any}(0)
+for rT in Rotations.RotTypeList
+    if (length(rT.parameters) == 2)
+
+        # get the super type for the order parameter
+        order_type = super(Rotations.euler_order(rT))
+        for order in subtypes(order_type)
+            push!(rot_types, rT{order})
+        end
+    end
+
+    # ordered ones should have defaults so leave them in
+    push!(rot_types, rT)
+end
+
+# define null rotations for conveniences
+null_rotation(::Type{RotMatrix}) = RotMatrix(eye(3))
+null_rotation(::Type{Quaternion}) = Quaternion(1.0,0,0,0)
+null_rotation(::Type{AngleAxis}) = AngleAxis(0.0, 1.0, 0, 0)
+null_rotation(::Type{SpQuat}) = SpQuat(0.0,0.0,0)
+null_rotation(::Type{EulerAngles}) = EulerAngles(0.0,0.0,0)
+null_rotation(::Type{ProperEulerAngles}) = ProperEulerAngles(0.0,0.0,0)
+null_rotation{ORD}(::Type{EulerAngles{ORD}}) = EulerAngles{ORD, Float64}(0.0,0.0,0)
+null_rotation{ORD}(::Type{ProperEulerAngles{ORD}}) = ProperEulerAngles{ORD, Float64}(0.0,0.0,0)
```

Comment:
Perhaps these should be public methods (`eye` would be somewhat consistent as a name, though consistency is the only real thing it's got going for it...)  `eye(RotMatrix{Float64})` will already work btw.